test/egraphs/ematch.jl

```diff
@@ -13,56 +13,56 @@ b = OptBuffer{UInt128}(10)
   r = @rule 2 --> true
   g = EGraph(2)
 
-  @test r.ematcher_left!(g, 0, g.root, r.stack, b) == 1
+  @test r.ematcher_left!(g, 0, g.root, get_local_stack(), b) == 1
```

Comment:
We could move all the calls to a file top-level `const stack = get_local_stack()`

Reply:
That doesn't seem right... are constants really not shared between threads? I feel like the RNG in Base wouldn't work this way if that were true.

Reply:
Yeah:

```julia
const r = Ref(2)

using Base.Threads

Threads.@threads for i = 1:10
  r[] += 1
end

r[] # 12
```

Reply:
Yeah but these tests are going to be run sequentially, but it's ok like this